packages/typespec-client-generator-core/test/packages/streams.test.ts

```diff
@@ -0,0 +1,190 @@
+      `,
+      );
+      const sdkPackage = runner.context.sdkPackage;
+      const method = getServiceMethodOfClient(sdkPackage);
+      strictEqual(sdkPackage.models.length, 0);
+      strictEqual(method.parameters[1].type.kind, "bytes");
+      strictEqual(method.parameters[1].type.encode, "bytes");
+      deepStrictEqual(method.operation.bodyParam?.type, method.parameters[1].type);
+    });
+  });
+
+  describe("stream response", () => {
+    it("http stream response", async () => {
+      await runner.compileWithBuiltInService(
+        `
+        @route("/")
+        op get(): HttpStream<Thing, "application/jsonl", string>;
+
+        model Thing { id: string }
+      `,
+      );
+      const sdkPackage = runner.context.sdkPackage;
+      const method = getServiceMethodOfClient(sdkPackage);
+      strictEqual(sdkPackage.models.length, 0);
+      strictEqual(method.response.type?.kind, "bytes");
+      strictEqual(method.response.type?.encode, "bytes");
+      strictEqual(method.operation.responses.length, 1);
+      strictEqual(method.operation.responses[0].type?.kind, "bytes");
+      strictEqual(method.operation.responses[0].type?.encode, "bytes");
+    });
+
+    it("json stream response", async () => {
+      await runner.compileWithBuiltInService(
+        `
+        @route("/")
+        op get(): JsonlStream<Thing>;
+
+        model Thing { id: string }
+      `,
+      );
+      const sdkPackage = runner.context.sdkPackage;
+      const method = getServiceMethodOfClient(sdkPackage);
+      strictEqual(sdkPackage.models.length, 0);
+      strictEqual(method.response.type?.kind, "bytes");
+      strictEqual(method.response.type?.encode, "bytes");
+      strictEqual(method.operation.responses.length, 1);
+      strictEqual(method.operation.responses[0].type?.kind, "bytes");
+      strictEqual(method.operation.responses[0].type?.encode, "bytes");
+    });
+
+    it("custom stream response", async () => {
+      await runner.compileWithBuiltInService(
+        `
+        @streamOf(Thing)
+        model CustomStream {
+          @header contentType: "custom/built-here";
+          @body body: bytes;
+        }
+
+        @route("/")
+        op get(): CustomStream;
+
+        model Thing { id: string }
+      `,
+      );
+      const sdkPackage = runner.context.sdkPackage;
+      const method = getServiceMethodOfClient(sdkPackage);
+      strictEqual(sdkPackage.models.length, 0);
+      strictEqual(method.response.type?.kind, "bytes");
+      strictEqual(method.response.type?.encode, "bytes");
+      strictEqual(method.operation.responses.length, 1);
+      strictEqual(method.operation.responses[0].type?.kind, "bytes");
+      strictEqual(method.operation.responses[0].type?.encode, "bytes");
+    });
+
+    it("intersection stream response", async () => {
+      await runner.compileWithBuiltInService(
+        `
+        @route("/")
+        op get(): JsonlStream<Thing> & { @statusCode statusCode: 204; };
```

Comment:
what would the tcgc behavior be if it's ssestream?

Reply:
same behavior as json stream. let me add tests for sse.

Reply:
done